src/Test.Utilities/DiagnosticExtensions.cs

```diff
@@ -25,10 +27,12 @@ public static class DiagnosticExtensions
                 CompilationUtils.ValidateNoCompileErrors(diagnostics);
             }
 
-            var analyzerDiagnostics = compilationWithAnalyzers.GetAnalyzerDiagnosticsAsync().Result;
```

Comment:
Calling both GetAnalyzerDiagnosticsAsync() and GetAllDiagnosticsAsync() seemed to cause the weirdness I saw, where my analyzer's compilation end action and operation callbacks would be invoked twice with different Compilation objects, but the compilation start action would only be invoked once.

Reply:
Opened https://github.com/dotnet/roslyn/issues/35149